specification/columns/skuid.md

```diff
@@ -1,27 +1,27 @@
 # SKU ID
 
-A SKU ID is an unique identifier that defines a provider-supported construct for organizing properties that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the SKU. The composition of the properties associated with the SKU ID may differ across providers. Some providers may not support the [*SKU*](#glossary:sku) construct and instead associate all such properties directly with the *SKU Price*. SKU ID is commonly used for analyzing cost based on *SKU* related properties above the pricing constructs.
+A SKU ID is a unique identifier that defines a provider-supported construct for organizing properties that are common across one or more [*SKU Prices*](#glossary:sku-price). SKU ID can be referenced on a catalog or [*price list*](#glossary:price-list) published by a provider to look up detailed information about the SKU. The composition of the properties associated with the SKU ID may differ across providers. Some providers may not support the [*SKU*](#glossary:sku) construct and instead associate all such properties directly with the *SKU Price*. SKU ID is commonly used for analyzing cost based on *SKU*-related properties above the pricing constructs.
 
-The SkuId column MUST be present in the billing data. This column MUST be of type String. The SkuId MUST NOT be null when [SkuPriceId](#skupriceid) is not null. SkuId MUST equal SkuPriceId when a provider does not support an overarching SKU ID construct.
+The SkuId column MUST be present in the billing data when the provider publishes a SKU list. This column MUST be of type String. The SkuId MUST NOT be null when [SkuPriceId](#skupriceid) is not null. SkuId MUST equal SkuPriceId when a provider does not support an overarching SKU ID construct.
```

Comment:
Shouldn't it be required when a provider offers multiple SKUs? There are many companies who don't "publish" a list of SKUs as an API. But that shouldn't negate the importance of identifying which of their multiple SKUs they're being charged for, right?
```suggestion
The SkuId column MUST be present in the billing data when the provider supports multiple SKUs. This column MUST be of type String. The SkuId MUST NOT be null when [SkuPriceId](#skupriceid) is not null. SkuId MUST equal SkuPriceId when a provider does not support an overarching SKU ID construct.
```

Reply:
If a provider supports multiple SKUs, the provider would highly probably publish a SKU list, wouldn't he?
If a provider has a unique SKU, and if the provider publishes it, I find it good to see the provider unique SKU in FOCUS data, because the provider may plan to have more SKUs later, which would then appear in FOCUS data.

Reply:
I'm in the camp of if a provider has multiple SKUs and SKUPriceIds, AND customers need to understand those details, they will have to present a SKU / SKUPrice list to their customers (regardless of the billing data). If a provider doesn't give those details, there's an assumption that it's because their services don't need that level of detail in billing data (or things like ChargeDescription are sufficient). At that point, requiring this column means a provider now is putting random IDs that can't be referenced anywhere and therefore have no meaning.

Maybe the issue here is the word 'publish' which might hint it's public. Would something like 'makes a SKU list available' or 'presents a SKU list' be more agreeable?

Reply:
With the introduction of the concept of FOCUS essentials and the discussion in https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/331#discussion_r1531676285 and https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/331#discussion_r1531645744, I am starting to think that the sentences "The XXX column MUST be present in the billing data when the provider supports ..." are not really useful:
- FOCUS Essential columns must be present in any FOCUS dataset. We have selected FOCUS essential columns so that none should be a barrier for a provider.
- Any other column may be present or not in a FOCUS dataset. If present, the values must comply with the specification, in particular the cases where the value must not be null. I don't see and I wonder whether there is a case where a provider would support a column but would make it absent.

Reply:
Microsoft supports availability zones but doesn't have that in the cost data, so we might not include the column.

Reply:
"Publishes" and "makes available" both feel very ambiguous. I would imagine practitioners would want this to identify similar rows, regardless of whether they have the IDs or not. They can be learned based on this data. Having a separate list makes them even more useful, but there's value regardless.

Reply:
> Microsoft supports availability zones but doesn't have that in the cost data, so we might not include the column.

They could.. that would just not meet FOCUS compatibility.

Reply:
> * Any other column may be present or not in a FOCUS dataset. If present, the values must comply with the specification, in particular the cases where the value must not be null. I don't see and I wonder whether there is a case where a provider would support a column but would make it absent.

To be clear, this isn't meant to say they have the column but are hiding it. It's completely possible that a provider supports resources but doesn't provide resource-level billing. One major CSP was in this bucket till recently. The sentence is supposed to convey (maybe needs refining?) that when the provider's general capabilities support capability X (which is a concept in FOCUS), those relevant FOCUS columns must be made available.